yarl/_url.py

```diff
@@ -727,9 +758,7 @@ def _make_child(self, paths, encoded=False):
                     f"Appending path {path!r} starting from slash is forbidden"
                 )
             path = path if encoded else self._PATH_QUOTER(path)
-            segments = [
-                segment for segment in reversed(path.split("/")) if segment != "."
```

Comment:
Not sure how removing this condition relates to anything?

Reply:
Good catch.
Artifact from the future, was meant to be in the next patches addressing URL.join. Removing "." breaks URL.join using URL.joinpath, it basically removes an empty element.